service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/C21OrderController.java

```diff
@@ -0,0 +1,122 @@
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.document.domain.Document;
+import uk.gov.hmcts.reform.fpl.interfaces.C21CaseOrderGroup;
+import uk.gov.hmcts.reform.fpl.model.C21Order;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisDocument;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.service.CreateC21OrderService;
+import uk.gov.hmcts.reform.fpl.service.DocmosisDocumentGeneratorService;
+import uk.gov.hmcts.reform.fpl.service.UploadDocumentService;
+import uk.gov.hmcts.reform.fpl.service.ValidateGroupService;
+
+import java.util.Map;
+
+import static org.apache.commons.lang3.StringUtils.defaultIfBlank;
+import static uk.gov.hmcts.reform.fpl.enums.DocmosisTemplates.C21;
+
+@Api
+@RequestMapping("/callback/create-order")
+@RestController
+public class C21OrderController {
+
+    private final ObjectMapper mapper;
+    private final DocmosisDocumentGeneratorService docmosisService;
+    private final UploadDocumentService uploadDocumentService;
+    private final CreateC21OrderService createC21OrderService;
+    private final ValidateGroupService validateGroupService;
+
+    @Autowired
+    public C21OrderController(ObjectMapper mapper,
+                              DocmosisDocumentGeneratorService docmosisService,
+                              UploadDocumentService uploadDocumentService,
+                              CreateC21OrderService createC21OrderService,
+                              ValidateGroupService validateGroupService) {
+        this.mapper = mapper;
+        this.docmosisService = docmosisService;
+        this.uploadDocumentService = uploadDocumentService;
+        this.createC21OrderService = createC21OrderService;
+        this.validateGroupService = validateGroupService;
+    }
+
+    @PostMapping("/about-to-start")
+    public AboutToStartOrSubmitCallbackResponse handleAboutToStart(@RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        CaseData caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
+
+        return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(caseDetails.getData())
+            .errors(validateGroupService.validateGroup(caseData, C21CaseOrderGroup.class))
+            .build();
+    }
+
+    @PostMapping("/mid-event")
+    public AboutToStartOrSubmitCallbackResponse handleMidEvent(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody CallbackRequest callbackRequest) {
+        CaseDetails caseDetails = callbackRequest.getCaseDetails();
+        Map<String, Object> data = caseDetails.getData();
+        CaseData caseData = mapper.convertValue(data, CaseData.class);
+
+        String index = (caseData.getC21OrderBundle() != null)
```

Comment:
Do you think this should be a helper method?

Reply:
Extracted into service 👍